### PR TITLE
06. Test "Categories" dropdown

### LINKED ISSUE #7: 06. Test "Categories" dropdown

### PR DESCRIPTION
**Pull request by maksym-bielyshev**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@pytest.fixture(scope="function")
+def init_driver(request: FixtureRequest) -> None:
+    """Remote driver initialization with required options.
+
+    :param request: FixtureRequest
+    :return: None
+    """
+    options = Options()
+    driver = webdriver.Remote(
+        command_executor='http://127.0.0.1:4444/wd/hub',
```
Comment (denyskulykov) [inline tests/conftest.py]: Change to local

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    yield driver
+
+
+def get_test_data(file_name: str) -> list:
+    """Converts file with test data to tuples list to use it in parameterized tests.
+    :param file_name: string
+    :return: test_data_list consists of tuples, where each tuple is one file row.
+    """
+    with open(f'../tests_data/{file_name}', 'r', encoding="utf8") as file:
+        reader = csv.reader(file, quoting=csv.QUOTE_ALL, skipinitialspace=True, delimiter='\t')
```
Comment (denyskulykov) [inline tests/conftest.py]: Try to read by sell not line
Мото uk/moto/?page=1

File: pages/base_page.py [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
         """Initialize main class.
 
         :param driver: Remote.
         """
         self._driver = driver
+
+    def switch_to_ukrainian(self):
+        if "auto.ria.com/uk" not in self._driver.current_url:
+            self._driver.find_element_by_xpath(
+                "//a[@class='selectLang ml-r']").click()
```
Comment (denyskulykov) [inline pages/base_page.py]: Move to PageObject

File: tests_data/category_dropdown_without_naming.csv [hunk 4]
```diff
@@ -0,0 +1,4 @@
+ru;Любой;https://auto.ria.com/search/?indexName=auto,order_auto,newauto_search
```
Comment (denyskulykov) [inline tests_data/category_dropdown_without_naming.csv]: remove path of url  **https://auto.ria.com**

File: pages/base_page.py (hunk 3 above)
Comment (maksym-bielyshev) [reply pages/base_page.py]: Done

File: tests_data/category_dropdown_without_naming.csv (hunk 4 above)
Comment (maksym-bielyshev) [reply tests_data/category_dropdown_without_naming.csv]: Done

File: tests/conftest.py (hunk 1 above)
Comment (maksym-bielyshev) [reply tests/conftest.py]: Done

File: tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+
+@pytest.fixture(scope="function")
+def init_driver(request: FixtureRequest) -> None:
+    """Remote driver initialization with required options.
+
+    :param request: FixtureRequest
+    :return: None
+    """
+    driver = webdriver.Chrome('../driver/chromedriver_win32.exe')
```
Comment (denyskulykov) [inline tests/conftest.py]: Move drive's path to config

File: tests/test_home_page.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+class TestHomePage(BaseTest):
+    """Class for the 'Home' page of 'AUTO.RIA'"""
+
+    def setup(self) -> None:
+        """Setup for the each test.
+
+        :return: None
+        """
+        self.driver.get("https://auto.ria.com/uk/")
```
Comment (denyskulykov) [inline tests/test_home_page.py]: Move the url to config

File: tests/test_home_page.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        # self.home_page = HomePage(self.driver)
+        self.home_page.click_search_button()
+        assert substring in self.driver.current_url
+
+    @pytest.mark.parametrize(
+        'language, category, substring',
+        get_test_data('category_dropdown_without_naming.csv'))
+    def test_categories_dropdown_transition_without_naming(
+            self, language, category, substring) -> None:
+        """Check options in the "Categories" dropdown, which have no
```
Comment (denyskulykov) [inline tests/test_home_page.py]: typo - has

File: tests/test_home_page.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+        Steps:
+            1. Click on the "Categories" dropdown
+            2. Select an option
+            3. Click on the "Пошук" button
+        Expected result:
+            1. Link in the address bar is changed to the address of the
+               corresponding category.
+        """
+        BasePage(self.driver).switch_proper_language(language)
```
Comment (denyskulykov) [inline tests/test_home_page.py]: Change basepage to home page

File: tests/test_home_page.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            1. Link in the address bar is changed to the address of the
+               corresponding category.
+        """
+        home_page = open_home_page
+        home_page.switch_proper_language(language)
+        home_page.category_dropdown.choose_dropdown_option(category)
+        home_page.click_search_button()
+        home_page._driver.execute_script("return document.readyState")
+        import time
+        time.sleep(5)
```
Comment (denyskulykov) [inline tests/test_home_page.py]: Move to method click_search_button

File: tests/test_home_page.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+        Expected result:
+            1. Link in the address bar is changed to the address of the
+               corresponding category.
+        """
+        home_page = open_home_page
+        home_page.switch_proper_language(language)
+        home_page.category_dropdown.choose_dropdown_option(category)
+        home_page.click_search_button()
+        home_page._driver.execute_script("return document.readyState")
```
Comment (denyskulykov) [inline tests/test_home_page.py]: the same

**Review by denyskulykov — APPROVED** [review]

**Pull request merged**